src/main/java/randoop/reflection/OmitMethodsPredicate.java

```diff
@@ -0,0 +1,229 @@
+ * the raw signature of an operation, and (2) test the raw signature of an operation and that of the
+ * same operation in superclasses.
+ */
+public class OmitMethodsPredicate {
+
+  /** An OmitMethodsPredicate that does no omission. */
+  public static OmitMethodsPredicate NO_OMISSION = new OmitMethodsPredicate(null);
+
+  /** {@code Pattern}s that match operations that should be omitted. */
+  private final List<Pattern> omitPatterns;
+
+  /**
+   * If {@code omitPatterns} is null, treat it as the empty list.
+   *
+   * @param omitPatterns a list of regular expressions for method signatures, or null
+   */
+  public OmitMethodsPredicate(List<Pattern> omitPatterns) {
+    if (omitPatterns == null) {
+      this.omitPatterns = new ArrayList<>();
+    } else {
+      this.omitPatterns = new ArrayList<>(omitPatterns);
+    }
+  }
+
+  /**
+   * Returns true if the operation is a constructor or method call and some omit pattern matches the
+   * {@link RawSignature} of the operation.
+   *
+   * @param operation the operation to match against the omit patterns of this predicate
+   * @return true if the signature matches an omit pattern, and false otherwise
+   */
+  boolean shouldOmit(TypedClassOperation operation) {
+    // Done if there are no patterns
+    if (omitPatterns.isEmpty()) {
+      return false;
+    }
+    // Only match constructors or methods
+    if (!operation.isConstructorCall() && !operation.isMethodCall()) {
+      return false;
+    }
+
+    String signature = operation.getRawSignature().toString();
+
+    for (Pattern pattern : omitPatterns) {
+      boolean result = pattern.matcher(signature).find();
+      if (Log.isLoggingOn()) {
+        Log.logLine(
+            String.format(
+                "Comparing '%s' against pattern '%s' = %b%n", signature, pattern, result));
+      }
+      if (result) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Indicates whether an omit pattern matches the raw signature of the method in either the
+   * declaring class of the method or a supertype.
+   *
+   * <p>Needs to search all supertypes of {@code classType} that have a declared member
+   * corresponding to the method. The type {@code classType} is a subtype of or equal to the
+   * declaring class of the operation.
+   *
+   * <p>It is necessary to search all types from {@code classType} up to {@code
+   * operation.getDeclaringType()}, and also to search for superclasses and interfaces of the
+   * declaring class that have the method.
+   *
+   * @param classType the type to which the method belongs
+   * @param operation the operation for the method
+   * @param method the reflection object for the method
+   * @return true if the signature of the method in the current class or a super class matches an
+   *     omit pattern, false otherwise
+   */
+  boolean shouldOmit(
+      ClassOrInterfaceType classType, final TypedClassOperation operation, final Method method) {
```

Comment:
Why are all three arguments needed?

The TypedClassOperation has a class associated with it already.  Is that guaranteed to be the same as `classType`, or are they guaranteed to be different, or are they sometimes the same and sometimes different?

I have the same questions about the relationship to `method`.

Please document the relationship among the arguments and what each one is used for.